pkg/agent/secondarynetwork/podwatch/controller.go

```diff
@@ -625,6 +643,115 @@ func (pc *PodController) initializeSecondaryInterfaceStore() error {
 	return nil
 }
 
+// initializeSRIOVSecondaryInterfaceStore restores secondary interfaceStore for SR-IOV interfaces
+// when agent restarts. It will get the Pod info from the store of primary interfaces, and check
+// the NetworkStatus annotation of a Pod, then restore the SR-IOV interfaces based on the NetworkAttachmentDefinition
+// name and device's pci_address in the NetworkStatus.
+func (pc *PodController) initializeSRIOVSecondaryInterfaceStore() error {
+	knownInterfaces := pc.primaryInterfaceStore.GetInterfacesByType(interfacestore.ContainerInterface)
+	for _, ifconf := range knownInterfaces {
+		podNamespace := ifconf.ContainerInterfaceConfig.PodNamespace
+		podName := ifconf.ContainerInterfaceConfig.PodName
+		pod, err := pc.podLister.Pods(podNamespace).Get(podName)
+		if err != nil {
+			klog.ErrorS(err, "Failed to get Pod", "Pod", klog.KRef(podNamespace, podName))
+			continue
+		}
+		objRef := klog.KObj(pod)
+		_, found := checkForPodSecondaryNetworkAttachment(pod)
+		if !found {
+			klog.V(2).InfoS("Pod does not have a NetworkAttachmentDefinition", "Pod", objRef)
+			continue
+		}
+		netStatus, err := netdefutils.GetNetworkStatus(pod)
+		if err != nil {
+			klog.ErrorS(err, "Failed to get NetworkStatus for Pod", "Pod", objRef)
+			continue
+		}
+
+		cache, err := pc.buildVFDeviceIDListPerPod(podName, podNamespace)
+		if err != nil {
+			return err
+		}
+
+		podKey := podKeyGet(podName, podNamespace)
+		var sriovInfo []podSriovVFDeviceIDInfo
+		for _, status := range netStatus {
+			if status.DeviceInfo == nil {
+				continue
+			}
+			resourceName, err := getNADResourceName(pc.netAttachDefClient, pod, status.Name)
```

Comment:
could we list all NADs first (out of the loop), then pass the NAD to the function to avoid repeatedly querying the same NAD for each Pod?

Reply:
Updated.